data/moves.ts

```diff
@@ -14916,10 +14916,20 @@ export const Moves: import('../sim/dex-moves').MoveDataTable = {
 					}
 					// Run through each action in queue to check if the Pursuit user is supposed to Mega Evolve this turn.
 					// If it is, then Mega Evolve before moving.
-					if (source.canMegaEvo || source.canUltraBurst) {
+					if (source.canMegaEvo || source.canUltraBurst || source.canTerastallize) {
 						for (const [actionIndex, action] of this.queue.entries()) {
-							if (action.pokemon === source && action.choice === 'megaEvo') {
-								this.actions.runMegaEvo(source);
+							if (action.pokemon === source) {
+								switch (action.choice) {
+								case 'megaEvo':
+									this.actions.runMegaEvo(source);
+									break;
+								// Also a "forme" change that happens before moves, though only possible in NatDex
+								case 'terastallize':
+									this.actions.terastallize(source);
+									break;
+								default:
+									continue;
+								}
```

Comment:
```suggestion
							if (action.pokemon === source) {
								if (action.choice === 'megaEvo') {
									this.actions.runMegaEvo(source);
								} else if (action.choice === 'terastallize') {
									// Also a "forme" change that happens before moves, though only possible in NatDex
									this.actions.terastallize(source);
								}
```

Reply:
Doesn't this change break the case where there's a different action, or does that never happen?